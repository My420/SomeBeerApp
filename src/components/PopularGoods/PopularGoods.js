import React from 'react';
import styles from './PopularGoods.module.scss';
import PopularBeerList from '../PopularBeer/PopularBeer';

const PopularGoods = () => {
  // eslint-disable-next-line no-console
  console.log('render ===== PopularGoods');
  return (
    <section className={styles.popular}>
      <h2 className={styles.title}>Most popular</h2>
      <PopularBeerList />
    </section>
  );
};

export default PopularGoods;
