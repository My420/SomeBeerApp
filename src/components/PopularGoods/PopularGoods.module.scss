.popular {
  padding-top: 40px;
}

.title {
  text-align: center;
  text-transform: uppercase;
  margin-bottom: 20px;
}
