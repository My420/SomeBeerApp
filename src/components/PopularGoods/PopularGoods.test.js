import React from 'react';
import { shallow } from 'enzyme';
import PopularGoods from './PopularGoods';

describe('Test <PopularGoods /> component', () => {
  const wrapper = shallow(<PopularGoods />);

  test('should correct render <PopularGoods /> component ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <section> tag with className = popular', () => {
    expect(wrapper.find('section').prop('className')).toBe('popular');
  });

  test('should render <h2> tag with className = title inside <section/>', () => {
    expect(
      wrapper
        .find('section')
        .find('h2')
        .prop('className')
    ).toBe('title');
  });

  test('should render <PopularBeerList/> component inside <section/>', () => {
    expect(wrapper.find('section').find('Connect(PopularBeer)')).toHaveLength(
      1
    );
  });
});
