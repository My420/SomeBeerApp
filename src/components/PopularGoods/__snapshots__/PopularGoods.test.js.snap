// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PopularGoods /> component should correct render <PopularGoods /> component  1`] = `
<section
  className="popular"
>
  <h2
    className="title"
  >
    Most popular
  </h2>
  <Connect(PopularBeer) />
</section>
`;
